tiatoolbox/models/architecture/hovernet_plus.py

```diff
@@ -0,0 +1,238 @@
+                    ),
+                    (
+                        "ls",
+                        HoVerNet._create_decoder_branch(ksize=ksize, out_ch=num_layers),
+                    ),
+                ]
+            )
+        )
+
+        self.upsample2x = UpSample2x()
+
+    @staticmethod
+    def _proc_ls(ls_map: np.ndarray):
+        """Extract Layer Segmentation map with LS Map.
+
+        This function takes the layer segmentation map and applies a gaussian blur to
+        remove spurious segmentations.
+
+        Args:
+            ls_map: The input predicted segmentation map.
+
+        Returns:
+            ls_map: The processed segmentation map.
+
+        """
+        ls_map = np.squeeze(ls_map.astype("float32"))
+        ls_map = cv2.GaussianBlur(ls_map, (7, 7), 0)
+        ls_map = np.around(ls_map)
+        ls_map = ls_map.astype("int")
+
+        return ls_map
+
+    @staticmethod
+    def _get_layer_info(pred_layer):
+        """Transforms image layers/regions into contours to store in dictionary.
+
+        Args:
+            image (ndarray): Semantic segmentation map of different
+                layers/regions following processing.
+
+        Returns:
+            layer_info_dict (dict): Dictionary to store layer contours in. It
+                has the following form:
+                layer_info = {
+                        contour: number[][],
+                        type: number,
+                }
+                layer_dict = {[layer_uid: number] : layer_info}
+
+        """
+
+        layer_list = np.unique(pred_layer)
+        layer_list = np.delete(layer_list, np.where(layer_list == 0))
+        layer_info_dict = {}
+        count = 1
+
+        for type_class in layer_list:
+            layer = np.where(pred_layer == type_class, 1, 0).astype("uint8")
+            contours, _ = cv2.findContours(
+                layer.astype("uint8"), cv2.RETR_TREE, cv2.CHAIN_APPROX_NONE
+            )
+            for layer in contours:
+                coords = layer[:, 0, :]
+                layer_info_dict[count] = {
+                    "contours": coords,
+                    "type": type_class,
+                }
+                count += 1
+
+        return layer_info_dict
+
+    @staticmethod
+    # skipcq: PYL-W0221
+    def postproc(raw_maps: List[np.ndarray]):
+        """Post processing script for image tiles.
+
+        Args:
+            raw_maps (list(ndarray)): list of prediction output of each head and
+                assumed to be in the order of [np, hv, tp, ls] (match with the output
+                of `infer_batch`).
+
+        Returns:
+            inst_map (ndarray): pixel-wise nuclear instance segmentation
+                prediction.
+            inst_dict (dict): a dictionary containing a mapping of each instance
+                within `inst_map` instance information. It has the following form
+            >>> inst_info = {
```

Comment:
Examples:

Reply:
Same as previous comment - this is the form of the output?

Reply:
Added example.